### DISCUSSION TITLE
Heiman Smoke Sensors different periodic status messages

### DISCUSSION THREAD

**Discussion by TheNutHouse84** [Koenkk/zigbee2mqtt#5727] [Devices]
I have 3 Heiman HS1SA-N smoke sensors. They all were ordered from the same vendor/same time. The device info page reports the same firmware date and build for all three. One sensor sends a status message that only has the battery and link quality. The other two send a more detailed message. See below. Could this be caused by difference in the reporting configuration sent to the device during pairing? Can this be fixed so the one sensor sends the more detailed status message? I don't have any device specific settings in configuration.yaml. New bare metal install on a Pi3B+ latest raspbian, current version of Z2M, Zstack3 on cc2530+cc2591 coord.

Two sensors send the following detailed message:
DEBUG: Received Zigbee message from '0x00158d000275592b', type 'attributeReport', cluster 'genPowerCfg', data '{"batteryPercentageRemaining":200}' from endpoint 1 with groupID 0
INFO: MQTT publish: topic 'z2m/0x00158d000275592b', payload '{"ac_status":false,"battery":100,"battery_low":false,"enrolled":false,"last_seen":"2021-01-12T09:03:39-05:00","linkquality":85,"restore_reports":true,"smoke":false,"supervision_reports":false,"tamper":false,"trouble":false}'

One sensor sends incomplete status message:
DEBUG: Received Zigbee message from '0x00158d000292eb69', type 'attributeReport', cluster 'genPowerCfg', data '{"batteryPercentageRemaining":153}' from endpoint 1 with groupID 0
INFO: MQTT publish: topic 'z2m/0x00158d000292eb69', payload '{"battery":76.5,"last_seen":"2021-01-12T09:30:01-05:00","linkquality":85}'

**Comment by sjorge** [comment]
Did you happen to hit the test button on the one sending the detailed message?

Most of those (ac_status, enrolled, ...) get added for me after pressing the do a test

**Reply by TheNutHouse84** [reply]
Yes I paired all 3 at approximately the same time using same process. I have also changed the reporting interval several times on all 3 to try to correct issue. I will try re-pairing and doing self test again just in case. Thanks

**Reply by TheNutHouse84** [reply]
Re-Pairing or doing a self test right after pairing  seems to have resolved the reporting differences. @sjorge Thanks for the insight.

**Reply by Morphy99** [reply]
It's fairly common occurrence for devices not to join completely sometimes in my experience. This can manifest itself with endpoints not being discovered or things not working correctly. It can usually be resolved by removing and adding again. With battery powered devices it can help to press a button every few seconds to make sure it keeps awake. The tricky part is only having one new device you've never used and then knowing when it hasn't joined correctly. Usually the first thing to try if you have any issues though.
